### PR TITLE
Use 10.15 as the deployment target for building with CMake

### PR DESCRIPTION
**Pull request by neonichu**

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Utilities/bootstrap [hunk 1]
```diff
@@ -517,6 +518,7 @@ def build_with_cmake(args, cmake_args, ninja_args, source_path, build_dir):
         ninja_cmd.append("-v")
 
 
+    call(["sed", "-i", "", "s/macosx10.10/macosx10.15/", "build.ninja"], cwd=build_dir)
```
Comment (neonichu) [inline Utilities/bootstrap:521]: This seems a bit hacky, but it looks like `MACOSX_DEPLOYMENT_TARGET` isn't fully plumbed through to the ninja generator of CMake or something? cc @compnerd

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test windows

**Review by tomerd — APPROVED** [review]

**Pull request merged**